### PR TITLE
[17.0][MIG] base_geoengine: Migration to 17.0 

### PR DESCRIPTION
**Pull request by peluko00**
Module migrated to version 17.0

cc https://github.com/APSL 154489
@miquelalzanillas @lbarry-apsl @javierobcn @mpascuall please review

- [Issue] https://github.com/OCA/geospatial/issues/349

To do:

- [ ] As an admin, if I change the domain of a layer with the filter button then the change are persisted in database. 
- [ ] As a user, if I change the domain of a layer with the filter button then the change are not persisted in database. There are just the changes in the display.
- [ ] As an admin,If the geoengine view is in edit mode, I can create new records by drawing them in the view.

### CODE REVIEW

**Comment by miquelalzanillas** [comment]
Hi @pedrobaeza and @reichie020212,

We used some code from @reichie020212 previous closed migration PR (https://github.com/OCA/geospatial/pull/357) and co-authored the migration commit.

Is this correct?

**Comment by pedrobaeza** [comment]
Yes, it seems OK. Please put the module name in the migration commit.

**Comment by miquelalzanillas** [comment]
> Yes, it seems OK. Please put the module name in the migration commit.

Done. Thanks!

**Comment by reichie020212** [comment]
hi @miquelalzanillas , Thank you for making me co-author.

**Comment by sersanchus** [comment]
I have noticed some issue in the JavaScript code of the controller in the modelParams function. It uses `this.props.state.modelState` twice without a `?` operator and in my tests on base_geoengine_demo `this.props.state` comes undefined.
Other controllers like in form, list or kanban are using a `?` also.

**Comment by sersanchus** [comment]
Some other things that I have noticed that need to be solved:

- Gist indexes are not created even if requested. This was fixed in 12.0 but not in subsequent ones.
- In the widget component, reprojection using 'ol.format.GeoJSON' is not well defined. So other srid than 3857 cannot be used. 
- The RecordsPanel is currently commented.

**Comment by peluko00** [comment]
Can you take a look and test it please @sersanchus

**Comment by sersanchus** [comment]
I'm actively working on this version at the moment and everything looks fine except for two things that I will try to solve ~~this week~~ today:

- The id associated with 'ol_map' is not enough to identify it with 'Date.now()' because the same form view can have several ending up with the same id.
- Recover and complete the RecordsPanel.

**Review by miquelalzanillas — APPROVED** [review]
Tested functionally from https://github.com/OCA/geospatial/pull/372. LGTM.

**Comment by miquelalzanillas** [comment]
> Tested functionally from #372. LGTM.

Sorry but I didn't tested following the 'base_geoengine_demo' readme and trying to follow those indications still are some bugs.

**Comment by sersanchus** [comment]
I just realized that the custom domain selector for geometries is completely broken. Odoo 17 doesn't allow overloading domain selection of custom field types as easily as before with:

`registry.category("domain_selector/fields")`

There is going to be a lot of work in that part.

**Review by mostafabarmshory — APPROVED** [review]

**Comment by mostafabarmshory** [comment]
Any update?

@peluko00

**Comment by peluko00** [comment]
> Any update?
> 
> @peluko00

I'll continue with this module maybe the next month

**Comment by mostafabarmshory** [comment]
@peluko00 

Tx to your commit. Seams everything is ok to merge, is`n it?

**Comment by peluko00** [comment]
> @peluko00
> 
> Tx to your commit. Seams everything is ok to merge, is`n it?

Not yet, they are some bugs to solve

**Comment by mostafabarmshory** [comment]
@peluko00 

Your last commit fix unit test. So we are ready to merge with the branch.

Is there any bug or issues?

I check it with odoo 17 manually and every thing seams ok. 

Let me know if there is any issue, pleas.

BTW, I can help you to finish it,


TX to your work, hope be fine

**Comment by peluko00** [comment]
> @peluko00
> 
> Your last commit fix unit test. So we are ready to merge with the branch.
> 
> Is there any bug or issues?
> 
> I check it with odoo 17 manually and every thing seams ok.
> 
> Let me know if there is any issue, pleas.
> 
> BTW, I can help you to finish it,
> 
> TX to your work, hope be fine

Yes, the bugs are in the TODO description of this PR. If you wanna to solve that make a pr to merge into my branch. Thanks in advance

File: base_geoengine/static/src/js/views/geoengine/geoengine_controller/geoengine_controller.esm.js [hunk 1]
```diff
@@ -40,24 +44,85 @@ export class GeoengineController extends Component {
                 offset: offset,
                 limit: limit,
                 total: count,
-                onUpdate: async ({offset, limit}) => {
```
Comment (sersanchus) [inline base_geoengine/static/src/js/views/geoengine/geoengine_controller/geoengine_controller.esm.js]: There might be a mistake here, the parameter names must be the same that odoo is using: offset and limit. If they don't match the pager can't be updated accordingly.
I could swear this was working when I checked it, has it been changed recently?

**Review by sersanchus — CHANGES_REQUESTED** [review]

File: base_geoengine/static/src/js/views/geoengine/geoengine_controller/geoengine_controller.esm.js (hunk 1 above)
Comment (peluko00) [reply base_geoengine/static/src/js/views/geoengine/geoengine_controller/geoengine_controller.esm.js]: Fixed, thanks!

**Review by sersanchus — APPROVED** [review]

**Comment by peluko00** [comment]
@dreispt can you review please

**Review by dreispt — APPROVED** [review]

**Pull request merged**